internal/p4/add.go

```diff
@@ -23,5 +24,14 @@ func (p *P4) Add(path string, opts ...Option) error {
 			return fmt.Errorf("unrecognized option %s", o.String())
 		}
 	}
-	return p.sh.Cmdf(`%s add %s -f "%s"`, p.cmd(), strings.Join(args, " "), path).RunErr()
+
+	// write paths to disk to avoid command line character limit
+	file, err := os.CreateTemp("", "p4harmonize_add_*.txt")
+	if err != nil {
+		return err
```

Comment:
I don't know off hand what the errors from CreateTemp look like, but I suspect adding some context here would help, something like: `return fmt.Errorf("error creating temp file for pattern %s: %w", <pattern>, err)`